File: src/menu.cpp
#include "engine/import.hpp"
#include "engine/engine.hpp"

class menu {
public:
    menu() {
        loading_screen();
    }
private:
    engine engine;

    void loading_screen() {
        Button new_project(engine.new_rect(50, 300, 150, 45), GRAY, LIGHTGRAY, "New Project");
        Button open_project(engine.new_rect(50, 350, 150, 45), GRAY, LIGHTGRAY, "Open Project");

        while (!WindowShouldClose())
        {

            BeginDrawing();

                ClearBackground(RAYWHITE);
                DrawText("Welcome to codecrafty!", 400, 10, 30, BLACK);
                new_project.draw();
                open_project.draw();


            EndDrawing();

            if (new_project.is_pressed()) {
                engine.print("New Project");
            }
            else if (open_project.is_pressed()) {
                engine.print("Open Project");
            }

        }
    }
};

File: src/menu/menu.go
package menu

import (
	"fmt"

	bd "codecrafty/src/backend"

	"fyne.io/fyne/v2"
	"fyne.io/fyne/v2/container"
)

func Menu(w fyne.Window) {
	newproject := bd.NewButton(w, "New Project", 50, 50, 200, 50)
	openproject := bd.NewButton(w, "Open Project", 50, 500, 200, 50)

	newproject.IsPressed(newProject)
	openproject.IsPressed(openProject)

	content := container.NewWithoutLayout(newproject.Container(), openproject.Container())

	bd.DrawContainer(w, content)

}

func newProject() {
	fmt.Println("New Project")
}

func openProject() {
	fmt.Println("Open Project")
}
